### PR TITLE
Fixed some problems in group code found during review

### PR DESCRIPTION
**Pull request by ethouris**
Things found during the review:

1. Synchronization group data with the socket data at the caller side:
* wrong reaction on isn == 0 (0 is a perfectly legal sequence value), which likely was remaining after the fact that a freshly created group has recorded scheduling sequence undefined
* the right "undefined sequence" value is -1, but this no longer makes sense because the group is now always initialized the schedule sequence number due to another bugfix
* the intention was to synchronize the latency value for the group from the value taken from the first socket in case when the group didn't get this information yet. The condition to do it was then changed to simply checking if the group is empty. Also the synchronization done at the caller side has been changed to not perform the sequence number derivation, as it doesn't make sense here - now the ISN is created in the group and it's the socket deriving this value from group, not the other way around

2. The use of `getMasterData` has proven to be now informative only - therefore the whole call is put under ENABLE_HEAVY_LOGGING condition for clarity.

3. Comparison of the expected peer id in case when the group is not empty should result in rejection. This is a probable situation if the user has mistakenly made two connections within a group, but these endpoints do not designate the same application at the peer side. In this case the returned peer group ID will be different for the second member socket, and this means that the address of the second connection is wrong, stating that the first was right.

### CODE REVIEW

File: srtcore/api.cpp [hunk 1]
```diff
@@ -1244,6 +1244,7 @@ int CUDTUnited::groupConnect(CUDTGroup* pg, SRT_SOCKGROUPCONFIG* targets, int ar
     // connection is going to later succeed or fail (this will be
     // known in the group state information).
     bool block_new_opened = !g.m_bOpened && g.m_bSynRecving;
+    bool was_empty = g.empty();
```
Comment (maxsharabayko) [inline srtcore/api.cpp:1247]:
```suggestion
    const bool was_empty = g.empty();
```

File: srtcore/core.cpp [hunk 2]
```diff
@@ -1862,8 +1862,10 @@ size_t CUDT::fillHsExtGroup(uint32_t* pcmdspec)
     if (m_parent->m_IncludedGroup->synconmsgno())
         flags |= SRT_GFLAG_SYNCONMSG;
 
+#if ENABLE_HEAVY_LOGGING
+
     SRTSOCKET master_peerid;
-    IF_HEAVY_LOGGING(steady_clock::duration master_tdiff);
+    steady_clock::duration master_tdiff;
```
Comment (maxsharabayko) [inline srtcore/core.cpp:1868]: `master_tdiff can be local (moved inside further ifs so that it is not visible in the whole function).

Comment (maxsharabayko) [inline srtcore/core.cpp:1868]: Is it possible to move the whole code block, scoped into `#if ENABLE_HEAVY_LOGGING`, into the `getMasterData(..)` function. And have `debugMasterData(..)` function instead.

File: srtcore/core.cpp [hunk 3]
```diff
@@ -1862,28 +1862,15 @@ size_t CUDT::fillHsExtGroup(uint32_t* pcmdspec)
...
-        HLOGC(cnlog.Debug, log << CONID() << "FOUND GROUP MASTER LINK: peer=$" << master_peerid
-                << " - start time diff: " << FormatDuration<DUNIT_S>(master_tdiff));
-    }
-    // (this function will not fill the variables with anything, if no master is found)
+    // NOTE: this code remains as is for historical reasons.
+    // The initial implementation stated that the peer id be
+    // extracted so that it can be reported and possibly the
+    // start time somehow encoded and written into the group
+    // extension, but it was later seen not necessary. Therefore
+    // this code remains, but now it's informational only.
```
Comment (maxsharabayko) [inline srtcore/core.cpp:1870]: Is it valid now?

Comment (ethouris) [reply srtcore/core.cpp:1870]: Yes. This explains why this call is here.

**Pull request merged**